resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentServiceImpl.java

```diff
@@ -254,14 +246,27 @@ public class ResidentServiceImpl implements ResidentService {
 
 	@Value("${resident.service.unreadnotificationlist.id}")
 	private String unreadnotificationlist;
-	
+
+	private static String authTypes;
+
+	@Value("${auth.types.allowed}")
+	public void setAuthTypes(String authType) {
+		authTypes = authType;
+	}
+
 	public static Function<? super AuthTypeStatusDtoV2, ? extends String> AUTH_TYPE_FUNCTION = t -> {
-		if(t.getAuthSubType()!=null) {
-			return t.getAuthType()+AUTH_TYPE_SEPERATOR+t.getAuthSubType();
-		}
-		else {
-			return t.getAuthType();
+		String[] authTypesArray = authTypes.split(",");
+		List<String> authTypesAllowed = new ArrayList<>(Arrays.asList(authTypesArray));
+		for(String authType : authTypesAllowed) {
```

Comment:
No need to create a list above, you can iterate using the array itself.

Reply:
done.